src/queryanalysis/utils/utils.go

```diff
@@ -0,0 +1,272 @@
+		case "blockingSessions":
+			queries[i].Query = fmt.Sprintf(queries[i].Query, arguments.QueryCountThreshold, config.TextTruncateLimit)
+		default:
+			fmt.Println("Unknown query type:", queries[i].Type)
+		}
+	}
+
+	return queries, nil
+}
+
+func ExecuteQuery(arguments args.ArgumentList, queryDetailsDto models.QueryDetailsDto, integration *integration.Integration, sqlConnection *connection.SQLConnection) ([]interface{}, error) {
+	log.Debug("Executing query: %s", queryDetailsDto.Query)
+
+	rows, err := sqlConnection.Connection.Queryx(queryDetailsDto.Query)
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute query: %w", err)
+	}
+	defer rows.Close()
+	log.Debug("Query executed: %s", queryDetailsDto.Query)
+	return BindQueryResults(arguments, rows, queryDetailsDto, integration, sqlConnection)
+}
+
+// BindQueryResults binds query results to the specified data model using `sqlx`
+func BindQueryResults(arguments args.ArgumentList,
+	rows *sqlx.Rows,
+	queryDetailsDto models.QueryDetailsDto,
+	integration *integration.Integration,
+	sqlConnection *connection.SQLConnection,
+) ([]interface{}, error) {
+	results := make([]interface{}, 0)
+
+	for rows.Next() {
+		switch queryDetailsDto.Type {
+		case "slowQueries":
+			var model models.TopNSlowQueryDetails
+			if err := rows.StructScan(&model); err != nil {
+				fmt.Println("Could not scan row: ", err)
+				continue
+			}
+			AnonymizeQueryText(model.QueryText)
+
+			results = append(results, model)
+
+			// fetch and generate execution plan
+			if model.QueryID != nil {
+				GenerateAndIngestExecutionPlan(arguments, integration, sqlConnection, *model.QueryID)
+			}
+
+		case "waitAnalysis":
+			var model models.WaitTimeAnalysis
+			if err := rows.StructScan(&model); err != nil {
+				fmt.Println("Could not scan row: ", err)
+				continue
+			}
+			AnonymizeQueryText(model.QueryText)
+
+			results = append(results, model)
+		case "blockingSessions":
+			var model models.BlockingSessionQueryDetails
+			if err := rows.StructScan(&model); err != nil {
+				fmt.Println("Could not scan row: ", err)
+				continue
+			}
+			AnonymizeQueryText(model.BlockedQueryText)
+			AnonymizeQueryText(model.BlockingQueryText)
+			results = append(results, model)
+		default:
+			return nil, fmt.Errorf("%w: %s", ErrUnknownQueryType, queryDetailsDto.Type)
+		}
+	}
+	return results, nil
+}
+
+func GenerateAndIngestExecutionPlan(arguments args.ArgumentList,
+	integration *integration.Integration,
+	sqlConnection *connection.SQLConnection,
+	queryID models.HexString,
+) {
+	hexQueryID := string(queryID)
+	executionPlanQuery := fmt.Sprintf(config.ExecutionPlanQueryTemplate, min(config.IndividualQueryCountMax, arguments.QueryCountThreshold),
+		arguments.QueryResponseTimeThreshold, hexQueryID, arguments.FetchInterval, config.TextTruncateLimit)
+
+	var model models.ExecutionPlanResult
+
+	rows, err := sqlConnection.Connection.Queryx(executionPlanQuery)
```

Comment:
Here the comment regarding the query sanitization applies as well: why using `fmt.Sprintf` instead of the arguments of `Queryx`?

Reply:
we are using variables on top of the query and then using those variables inside the query. Apart from that, the variable we are using are all integers? shall we keep it in the tech debt sheet?